project/src/com/google/daggerquery/executor/models/Query.java

```diff
@@ -79,20 +86,107 @@ public Query(String typeName, String... parameters) {
   /**
    * Executes query on a {@link BindingGraph}.
    *
-   * <p>For all queries return a list with string. Strings content depends on a query name.
-   * <p>For `deps` queries each string represent exactly one dependency.
+   * <p>For all queries return a list with strings. The content of the strings depends on a query name.
+   *
+   * <ul>
+   * <li>For `deps` query each string represents exactly one dependency.
+   * <li>For `allpaths` query each string contains a path between {@code source} and {@code target} nodes.
+   * The connection between nodes is shown with the construction '->'. For example, one of the possible paths
+   * may look like this: "com.google.First -> com.google.Second -> com.google.Third".
+   * </ul>
    */
   public List<String> execute(BindingGraph bindingGraph) {
-    if (name.equals("deps")) {
-      String sourceNode = parameters[0];
-      if (!bindingGraph.getAdjacencyListMap().containsKey(sourceNode)) {
-        throw new IllegalArgumentException("Specified source node doesn't exist.");
+    switch (name) {
+      case DEPS_QUERY_NAME:
+        String sourceNode = parameters[0];
+
+        if (!bindingGraph.getAdjacencyListMap().containsKey(sourceNode)) {
+          throw new IllegalArgumentException("Specified source node doesn't exist.");
+        }
+
+        return bindingGraph.getAdjacencyListMap().get(sourceNode).getDependencyList()
+            .stream().map(Dependency::getTarget).sorted().collect(toList());
+      case ALLPATHS_QUERY_NAME:
+        String source = parameters[0];
+
+        if (!bindingGraph.getAdjacencyListMap().containsKey(source)) {
+          throw new IllegalArgumentException("Specified source node doesn't exist.");
+        }
+
+        String target = parameters[1];
+        Set<String> visitedNodes = new HashSet<>();
+        ImmutableList.Builder<Path> result = new ImmutableList.Builder<>();
+        Path path = new Path<>();
+
+        findAllPaths(source, target, path, bindingGraph, visitedNodes, result);
+        return result.build().stream().map(Path::toString).collect(toList());
+    }
+
+    throw new NotImplementedException();
+  }
+
+  /**
+   * Represents a path between nodes of type {@code NodeT}.
+   *
+   * <p>Supports operations of adding a new node to the back and removing the last node.
+   *
+   * <p>Uses delimiter '->' for constructing the string representation of all data.
+   */
+  private class Path<NodeT> {
```

Comment:
Please make this a `static` inner class, because it does not require access to state variables in the outer class.

Reply:
Alternatively it may make sense to put `Path` in a separate file from `Query`, if they are two different concepts.

Reply:
Thanks! I would prefer to save it as an inner class because it is closely related to the query's output (i.e. it's tested as a part of `Query` class and used only in it).